### PR TITLE
feat(mmetadata): Added ability to copy&paste metadata

### PR DESCRIPTION
**Pull request by pavlovicnemanja**
SDANSA-224 - Save / Restore metadata into photo uploader

### CODE REVIEW

File: scripts/apps/search/views/multi-image-edit.html [hunk 1]
```diff
@@ -87,6 +87,14 @@ <h3 class="sd-slide-in-panel__heading" translate>Metadata</h3>
                         ></div>
                     </div>
                 </div>
+                <div class="sd-slide-in-panel__footer">
```
Comment (petrjasek) [inline scripts/apps/search/views/multi-image-edit.html]: it's missing translate annotations

File: scripts/apps/search/views/multi-image-edit.html [hunk 2]
```diff
@@ -87,6 +87,14 @@ <h3 class="sd-slide-in-panel__heading" translate>Metadata</h3>
...
+                    <div class="btn-group--flex sd-margin-b--1">
+                        <button type="button" class="btn btn--hollow btn--primary btn--expanded" ng-click="copyMetadata(metadata)" ng-disabled="!isDirty()" translate>Copy metadata</button>
+                        <button type="button" class="btn btn--hollow btn--expanded" ng-click="pasteMetadata()" ng-disabled="!metadataFromStorage" translate>Paste metadata</button>
+                    </div>
+                    <sd-check ng-model="autoFill" ng-click="toggleAutofill()">Autofill</sd-check>
+                    <a ng-click="clearSavedMetadata()" class="text-link text-link--white pull-right" ng-show="metadataFromStorage">Clear saved metadata</a>
```
Comment (petrjasek) [inline scripts/apps/search/views/multi-image-edit.html]: there is missing translate

File: scripts/apps/search/MultiImageEdit.ts [hunk 3]
```diff
@@ -196,6 +196,8 @@ export function MultiImageEditDirective(asset, $sce) {
         templateUrl: asset.templateUrl('apps/search/views/multi-image-edit.html'),
         link: function(scope) {
             scope.trustAsHtml = $sce.trustAsHtml;
+            scope.metadataFromStorage = localStorage.getItem('metadata:items');
```
Comment (petrjasek) [inline scripts/apps/search/MultiImageEdit.ts]: would be good to define constants for those storage keys

File: scripts/apps/search/MultiImageEdit.ts [hunk 4]
```diff
@@ -204,6 +206,32 @@ export function MultiImageEditDirective(asset, $sce) {
...
+                localStorage.setItem('metadata:items', JSON.stringify(metadata));
+            };
+
+            scope.toggleAutofill = () => {
+                scope.autoFill = !scope.autoFill;
+                localStorage.setItem('metadata:autofill', scope.autoFill);
+            };
+
+            scope.pasteMetadata = () => {
+                scope.metadata = JSON.parse(localStorage.getItem('metadata:items'));
```
Comment (petrjasek) [inline scripts/apps/search/MultiImageEdit.ts]: you have `scope.metadataFromStorage` defined earlier, why not use it here? or why load it before?

File: scripts/apps/search/MultiImageEdit.ts [hunk 5]
```diff
@@ -204,6 +206,32 @@ export function MultiImageEditDirective(asset, $sce) {
...
+                _.forEach(scope.metadata, (metadata, key) => {
+                    scope.onChange(key);
+                });
+            };
+
+            scope.clearSavedMetadata = () => {
+                localStorage.removeItem('metadata:items');
+                delete scope.metadataFromStorage;
```
Comment (petrjasek) [inline scripts/apps/search/MultiImageEdit.ts]: seems like you only use this to indicate if it's empty, imo would be better to populate it like:
```
scope.metadataFromStorage = !!localStorage.getItem('metadata:items');
```
and here you can just set it to `false`, so it doesn't compare the whole metadata json just to check it's empty

Comment (pavlovicnemanja) [reply scripts/apps/search/MultiImageEdit.ts]: done 🙂

File: scripts/apps/search/MultiImageEdit.ts (hunk 4 above)
Comment (pavlovicnemanja) [reply scripts/apps/search/MultiImageEdit.ts]: I changed scope.metadataFromStorage to be just boolean, so now I use it just for checks

File: scripts/apps/search/MultiImageEdit.ts (hunk 3 above)
Comment (pavlovicnemanja) [reply scripts/apps/search/MultiImageEdit.ts]: Where should I define them? Should I put it on top like ```CONST metadataItems = "metadata:item"```?

Comment (petrjasek) [reply scripts/apps/search/MultiImageEdit.ts]: top of file or just top of link, doesn't matter. and should  be more like:
```
const METADATA_ITEMS = 'metadata:items';
const METADATA_AUTOFILL = 'metadata:autofill';
```

Comment (pavlovicnemanja) [reply scripts/apps/search/MultiImageEdit.ts]: done 😄

**Pull request merged**